lib/runTests.js

```diff
@@ -53,29 +53,44 @@ grammar.rules.forEach((rule) => {
     trace.filter.rules[rule.name] = true;
 });
 
-// // set up AST
-// const ast = new AST();
-// parser.ast = ast;
-// // define AST callbacks, see https://github.com/ldthomas/apg-js2-examples/blob/61a36fd963ba544c0630a533c053f60d18d878d2/ast/setup.js#L29-L74
-// config.rulenameTokens.forEach(
-//   (ruleName) => (ast.callbacks[ruleName] = rulenameTokenCB(ruleName))
-// );
-// config.matchTokens.forEach(
-//   (ruleName) => (ast.callbacks[ruleName] = matchTokenCB)
-// );
-//
-// function rulenameTokenCB(rulename) {
+// set up AST
+const ast = new AST();
+parser.ast = ast;
+// define callbacks, see https://github.com/ldthomas/apg-js2-examples/blob/61a36fd963ba544c0630a533c053f60d18d878d2/ast/setup.js#L29-L74
+// function ruleNameTokenCB(ruleName) {
 //   return (state, chars, phraseIndex, phraseLength, data) => {
-//     if (state === ids.SEM_PRE) data.push(rulename);
+//     if (state === ids.SEM_PRE) data.push(ruleName);
 //     return ids.SEM_OK;
 //   };
 // }
-//
-// function matchTokenCB(state, chars, phraseIndex, phraseLength, data) {
-//   if (state === ids.SEM_PRE)
-//     data.push(utils.charsToString(chars, phraseIndex, phraseLength));
-//   return ids.SEM_SKIP;
-// }
+function matchTokenCB(ruleName) {
+  return (state, chars, phraseIndex, phraseLength, data) => {
+    if (state === ids.SEM_PRE)
+      data.push(
+        `${ruleName}:${utils.charsToString(chars, phraseIndex, phraseLength)}`
+      );
+    return ids.SEM_SKIP;
+  };
+}
+//TODO: tokens for matching rule name?
+// config.expect.ruleNameTokens.forEach(
+//   (ruleName) => (ast.callbacks[ruleName] = ruleNameTokenCB(ruleName))
+// );
+grammar.rules.forEach((rule) => {
+  if (
+    //TODO: move this to config.yaml or even the test case files similar to Constraints?
```

Comment:
Pick those rules that occur in `tc.Expect`.

Reply:
Yes, test-case specific. Why not?

Reply:
Done